diff --git a/internal/demo/worker/workflow.go b/internal/demo/worker/workflow.go
index aff2b7a..777528b 100644
--- a/internal/demo/worker/workflow.go
+++ b/internal/demo/worker/workflow.go
@@ -21,8 +21,16 @@ func HelloWorld(ctx workflow.Context) (string, error) {
 	}
 
 	// Sleep for a while
-	if err := workflow.ExecuteActivity(ctx, Sleep, 60).Get(ctx, nil); err != nil {
-		return "", err
+	// TODO(jlegrone): Remove version check after no workflows match the query:
+	//                 TemporalChangeVersion NOT IN ("sleep-without-activity-1") AND ExecutionStatus = "Running"
+	if v := workflow.GetVersion(ctx, "sleep-without-activity", workflow.DefaultVersion, 1); v == 1 {
+		if err := workflow.Sleep(ctx, time.Minute); err != nil {
+			return "", err
+		}
+	} else {
+		if err := workflow.ExecuteActivity(ctx, Sleep, 60).Get(ctx, nil); err != nil {
+			return "", err
+		}
 	}
 
 	// Return the greeting
